rpc/src/calls/personal.rs

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright 2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ------------------------------------------------------------------------------
+ */
+
+use accounts::{get_key_dir, Account};
+use client::BlockKey;
+use client::ValidatorClient;
+use jsonrpc_core::{Error, Params, Value};
+use messages::seth::{
+    CreateExternalAccountTxn, EvmPermissions, SethTransaction as SethTransactionPb,
+    SethTransaction_TransactionType,
+};
+use openssl::ec::{EcGroup, EcKey};
+use openssl::nid::Nid;
+use openssl::symm::Cipher;
+use requests::RequestHandler;
+use sawtooth_sdk::messaging::stream::MessageSender;
+use std::fs;
+use std::str::from_utf8;
+use transactions::SethTransaction;
+use transform;
+
+pub fn get_method_list<T>() -> Vec<(String, RequestHandler<T>)>
+where
+    T: MessageSender,
+{
+    vec![
+        ("personal_listAccounts".into(), list_accounts),
+        ("personal_newAccount".into(), new_account),
+        ("personal_unlockAccount".into(), unlock_account),
+        ("personal_importRawKey".into(), import_raw_key),
+    ]
+}
+
+macro_rules! except {
+    ($expr:expr, $msg:expr) => {{
+        $expr.map_err(|err| {
+            error!("{}", err);
+            Error {
+                code: (-32069i64).into(),
+                message: $msg.into(),
+                data: None,
+            }
+        })?
+    }};
+}
+
+#[allow(needless_pass_by_value)]
+pub fn list_accounts<T>(_: Params, client: ValidatorClient<T>) -> Result<Value, Error>
+where
+    T: MessageSender,
+{
+    info!("personal_listAccounts");
+
+    let accounts = client.loaded_accounts();
+    let locked_accounts = accounts.read().unwrap();
+
+    Ok(Value::Array(
+        locked_accounts
+            .iter()
+            .map(|a| transform::hex_prefix(a.address()))
+            .collect::<Vec<_>>(),
+    ))
+}
+
+#[allow(needless_pass_by_value)]
+pub fn new_account<T>(params: Params, client: ValidatorClient<T>) -> Result<Value, Error>
+where
+    T: MessageSender,
+{
+    info!("personal_newAccount");
+
+    let args: Vec<Option<String>> = match params.parse::<Vec<Option<String>>>() {
+        Ok(t) => t,
+        Err(_) => {
+            return Err(Error::invalid_params(
+                "Takes [password: String, moderator: String, permissions: String]",
+            ));
+        }
+    };
+
+    // We get Option<Option<String>> here due the possibility of a client
```

Comment:
I think this a typo and should say `Vec<Option<String>>`

Reply:
We have `Vec<Option<String>>`, but using `.get()` wraps it's response in an `Option` as well, returning `Option<Option<String>>`.